allennlp/models/multitask.py

```diff
@@ -0,0 +1,196 @@
+            if self._active_heads is not None and head_name not in self._active_heads:
+                continue
+
+            combined_arguments = {**backbone_outputs, **kwargs}
+            head_arguments = self._get_arguments(combined_arguments, head_name)
+
+            if (
+                self._active_heads is None
+                and self.training
+                and head_arguments.keys() != self._allowed_arguments[head_name]
+            ):
+                continue
+
+            head_outputs = self._heads[head_name](**head_arguments)
+            for key in head_outputs:
+                outputs[f"{head_name}_{key}"] = head_outputs[key]
+
+            if "loss" in head_outputs:
+                head_loss = self._loss_weights[head_name] * head_outputs["loss"]
+                if loss is None:
+                    loss = head_loss
+                else:
+                    loss += head_loss
+
+        if loss is not None:
+            outputs["loss"] = loss
+
+        return outputs
+
+    def _get_arguments(self, available_args: Dict[str, Any], component: str) -> Dict[str, Any]:
+        """
+        Given a list of things we might want to pass to a component (where "component" is either the
+        backbone or a head), this method figures out which things we should actually pass, by
+        mapping names and looking at allowed arguments.
+        """
+        allowed_args = self._allowed_arguments[component]
+        name_mapping = self._arg_name_mapping[component]
+        kept_arguments = {}
+        for key, value in available_args.items():
+            new_key = name_mapping.get(key, key)
+            if new_key in allowed_args:
+                if new_key in kept_arguments:
+                    raise ValueError(
+                        f"Got duplicate argument {new_key} for {component}. This likely means that"
+                        " you mapped multiple inputs to the same name. This is generally ok for"
+                        " the backbone, but you have to be sure each batch only gets one of those"
+                        " inputs.  This is typically not ok for heads, and means something is not"
+                        " set up right."
+                    )
+                kept_arguments[new_key] = value
+        return kept_arguments
+
+    @overrides
+    def get_metrics(self, reset: bool = False) -> Dict[str, float]:
+        metrics = {}
+        for head_name in self._heads:
+            for key, value in self._heads[head_name].get_metrics(reset).items():
+                metrics[f"{head_name}_{key}"] = value
+        return metrics
+
+    @overrides
+    def make_output_human_readable(
+        self, output_dict: Dict[str, torch.Tensor]
+    ) -> Dict[str, torch.Tensor]:
+        output_dict = self._backbone.make_output_human_readable(output_dict)
+        for head_name in self._heads:
+            if self._active_heads is not None and head_name not in self._active_heads:
+                continue
```

Comment:
If this respects `_active_heads`, then `get_metrics()` should do so too.

Reply:
Yes, good catch, thanks.